govuk_chat_evaluation/rag_answers/deepeval_evaluate.py

```diff
@@ -128,10 +128,11 @@ def convert_deepeval_output_to_evaluation_results(
                         RunMetricOutput(
                             run=run_idx,
                             metric=metric_data.name,
-                            score=metric_data.score,
-                            reason=metric_data.reason,
-                            cost=metric_data.evaluation_cost,
-                            success=metric_data.success,
+                            score=getattr(metric_data, "score", None),
+                            reason=getattr(metric_data, "reason", None),
+                            cost=getattr(metric_data, "evaluation_cost", None),
+                            success=getattr(metric_data, "success", None),
+                            error=getattr(metric_data, "error", None),
```

Comment:
Are you sure we need this change? If I understand right these are expected to be instances of [MetricData](https://github.com/confident-ai/deepeval/blob/e538925453bc50b24ff6c9d6053bf7d888166028/deepeval/tracing/api.py#L29-L39) 

I raise it because we should probably also add a test to flag it can handle a partial metric, but I'm not sure that scenario is possible?

Reply:
Thanks for that break down - sounds quite the rabbit hole 🐰 

I'd have not thought that even with the exclude_none option we'd have a problem if they were populating MetricData objects as they'd automatically set score to None if the field is ommited.

I'd suggest removing the fix here and seeing if you still get the issue. If you do get the issue again let's look into the cause further and see if we need to write a bug report with Deepeval or whether we've misunderstood that the class is MetricData and we need to actually test with other types as well.

Reply:
Thanks Kev, makes sense. 

Just to clarify:
> I'd suggest removing the fix here and seeing if you still get the issue
with the "fix here" do you mean over defenisve `getattr` approach or would you prefer to just to drop the whole fix?

Reply:
Sorry for ambiguity

I suggest removing the defensive getattr approach